snapshot-labs/snapshot
# fix: limit parallel event queries, goerli block lookback

### Issues
We are seeing RPC provider errors when loading specific proposals that  use osnap. This is due to long lookbacks and many parallel network requests for event data.
![image](https://github.com/snapshot-labs/snapshot/assets/4429761/5b50d9e4-c578-4a47-9bc5-7768dbe39295)

Fixes #

### Changes 
*_(Briefly describe the changes made in this PR)_*

1. This introduces "bluebird promises", specifically the map function, which allows us to set a concurrency level. this is used in the most likely place where there are too many requests made in parallel using promise.all.
2. we further add deploy block numbers for goerli, in case the users safe is in goerli, previously this was starting a query at block 0.
3. we also fix the default "something went wrong" at the bottom of the transaction form in the proposal.
![image](https://github.com/snapshot-labs/snapshot/assets/4429761/d833816c-d124-4790-9c5c-c83c4f7a6c35)


### How to test
*_(Explain how the changes can be tested, including any required setup steps)_*

1. you can see a something went wrong error here: https://snapshot.org/#/frp.eth/proposal/0xfd99f0127621cb5b644114dc6195fc0bbd3da7e88b41cce409b5ed82214ce106
2. see in dev console `ERR_INSUFFICIENT_RESOURCES`
3. visit this same proposal on this build and see that it eventually loads without errors.

### To-Do
this may buy more time to develop a better, faster caching solution


### Self-review checklist

### Additional notes or considerations
*_(Include any other relevant information or context that may be helpful for the reviewer)_*
